### ISSUE TITLE
can't find out about wayward deployment

### ISSUE THREAD

**Issue by sosiouxme** [openshift/origin#2743]
Usually if your deployment goes south, you have *something* to look at. But not always it seems.

```
[joe@master beta4]$ osc status
In project OpenShift 3 Demo (demo)

service mysql (172.30.213.58:3306)
  mysql deploys mysql:latest
    #1 deployment waiting on image or update

hello-openshift deploys hello-openshift:latest
  #1 deployment waiting on image or update

To see more information about a Service or DeploymentConfig, use 'osc describe service <name>' or 'osc describe dc <name>'.
You can use 'osc get all' to see lists of each of the types described above.
[joe@master beta4]$ osc describe dc mysql
Name:           mysql
Created:        About an hour ago
Labels:         <none>
Latest Version: Not deployed
Triggers:       Config, Image(mysql@latest, auto=true)
Strategy:       Recreate
Template:
                        Selector:       deploymentconfig=mysql
                        Replicas:       1
                        Containers:
                                NAME    IMAGE           ENV
                                mysql   library/mysql
Latest Deployment:      <none>
No events.
[joe@master beta4]$ osc deploy mysql
mysql #1 deployment waiting on image or update
[joe@master beta4]$ osc get rc
CONTROLLER   CONTAINER(S)   IMAGE(S)   SELECTOR   REPLICAS
[joe@master beta4]$ osc get pods
POD       IP        CONTAINER(S)   IMAGE(S)   HOST      LABELS    STATUS    CREATED   MESSAGE
[joe@master beta4]$ 
```

I assume there's something wrong with my docker or SDN config or something, but I don't even know which node to look on. I have literally no information about what is going wrong. Is there a command for describing a deployment? Don't see a pod, obviously.

**Comment by sosiouxme** [comment]
While the issue at hand is "I don't have any information as an end user" it's worth noting the source of the particular problem, which is probably this error from openshift-master:

`factory.go:68] couldn't create initial Deployment for DeploymentConfig demo/demo/mysql:0: DeploymentConfig "mysql" is invalid: triggers[1].imageChange.tag: invalid value 'latest': no image recorded for demo/mysql:latest`

Which BTW was created with `osc new-app mysql` so this shouldn't be hard to reproduce.

**Comment by sosiouxme** [comment]
Of course, that works fine on an origin devenv. So I still don't know what the actual problem is.

**Comment by ironcladlou** [comment]
I believe the error you see in the logs is a result of #2819. I think the particular error in question should result in an event. Then, calls to `osc status` and `osc describe dc` (and the console) would have reported the error to you (the end user). The error wouldn't have been actionable in this case, but we should have told you what we knew.

**Comment by smarterclayton** [comment]
The race needs to be fixed, the warning is lower priority.

**Comment by ironcladlou** [comment]
Let's downgrade this issue to cover the event we should emit. #2819 covers the more serious bug which was exposed.

**Comment by abhgupta** [comment]
Lowering to P3 based on comments above from @ironcladlou and @smarterclayton
